client/src/app/pages/applications/applications-table-assessment/applications-table-assessment.tsx

```diff
@@ -499,39 +508,55 @@ export const ApplicationsTable: React.FC = () => {
   };
 
   const discardAssessmentRow = (row: Application) => {
-    setApplicationAssessmentToDiscard(row);
+    setApplicationAssessmentOrReviewToDiscard(row);
     setIsDiscardAssessmentConfirmDialogOpen(true);
   };
 
-  const discardAssessment = (application: Application) => {
-    Promise.all([
-      application.review ? deleteReview(application.review.id!) : undefined,
-    ])
-      .then(() => {
-        const assessment = getApplicationAssessment(application.id!);
-        return Promise.all([
-          assessment ? deleteAssessment(assessment.id!) : undefined,
-        ]);
-      })
-      .then(() => {
-        pushNotification({
-          title: t("toastr.success.assessmentDiscarded", {
-            application: application.name,
-          }),
-          variant: "success",
-        });
+  const onDeleteReviewSuccess = (name: string) => {
+    pushNotification({
+      title: t("toastr.success.reviewDiscarded", {
+        application: name,
+      }),
+      variant: "success",
+    });
+    queryClient.invalidateQueries([ApplicationsQueryKey]);
+  };
 
-        queryClient.invalidateQueries([assessmentsQueryKey]);
-        queryClient.invalidateQueries([reviewsQueryKey]);
+  const onDeleteAssessmentSuccess = (name: string) => {
+    pushNotification({
+      title: t("toastr.success.assessmentDiscarded", {
+        application: name,
+      }),
+      variant: "success",
+    });
+    queryClient.invalidateQueries([ApplicationsQueryKey]);
+  };
 
-        fetchApplications();
-      })
-      .catch((error) => {
-        pushNotification({
-          title: getAxiosErrorMessage(error),
-          variant: "danger",
-        });
-      });
+  const onDeleteError = (error: AxiosError) => {
+    pushNotification({
+      title: `${error}`,
+      variant: "danger",
+    });
+  };
+
+  const { mutate: deleteReview } = useDeleteReviewMutation(
+    onDeleteReviewSuccess,
+    onDeleteError
+  );
+
+  const { mutate: deleteAssessment } = useDeleteAssessmentMutation(
+    onDeleteAssessmentSuccess,
+    onDeleteError
+  );
+
+  const discardAssessmentAndReview = (application: Application) => {
```

Comment:
Looking good with the fixes. I think QE will probably file another issue re: the notification for delete review & assessment will only show whichever of these api calls finishes first. This may require further rework to add a new mutation that wraps both the review & assessment delete.